src/html.rs

```diff
@@ -90,6 +97,11 @@ impl HTML {
             .read_from(&mut html.as_bytes())
             .map(|dom| Self::from_dom(dom, url))
     }
+
+    pub fn set_url(&mut self, url: Option<String>) {
```

Comment:
This should probably not be a public method right? There's no use for updating the URL when parsing is already done